modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbprocedure/XSKHDBProcedureParser.java

```diff
@@ -11,41 +11,100 @@
  */
 package com.sap.xsk.hdb.ds.parser.hdbprocedure;
 
-import java.sql.Timestamp;
-
+import com.sap.xsk.exceptions.XSKArtifactParserException;
+import com.sap.xsk.hdb.ds.api.IXSKDataStructureModel;
+import com.sap.xsk.hdb.ds.api.XSKDataStructuresException;
+import com.sap.xsk.hdb.ds.artefacts.HDBProcedureSynchronizationArtefactType;
 import com.sap.xsk.hdb.ds.model.XSKDataStructureParametersModel;
+import com.sap.xsk.hdb.ds.model.hdbprocedure.XSKDataStructureHDBProcedureModel;
+import com.sap.xsk.hdb.ds.parser.XSKDataStructureParser;
+import com.sap.xsk.hdb.ds.synchronizer.XSKDataStructuresSynchronizer;
+import com.sap.xsk.utils.XSKCommonsConstants;
+import com.sap.xsk.utils.XSKCommonsUtils;
 import com.sap.xsk.utils.XSKHDBUtils;
+import custom.HanaProcedureListener;
+import models.HDBProcedureDefinitionModel;
+import models.HDBProcedureMissingPropertyException;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
 import org.apache.commons.codec.digest.DigestUtils;
 import org.eclipse.dirigible.api.v3.security.UserFacade;
+import org.eclipse.dirigible.core.scheduler.api.ISynchronizerArtefactType;
 
-import com.sap.xsk.hdb.ds.api.IXSKDataStructureModel;
-import com.sap.xsk.hdb.ds.api.XSKDataStructuresException;
-import com.sap.xsk.hdb.ds.model.hdbprocedure.XSKDataStructureHDBProcedureModel;
-import com.sap.xsk.hdb.ds.parser.XSKDataStructureParser;
+import java.io.IOException;
+import java.sql.Timestamp;
+
+public class XSKHDBProcedureParser implements XSKDataStructureParser<XSKDataStructureHDBProcedureModel> {
+
+    private final XSKDataStructuresSynchronizer dataStructuresSynchronizer;
+    private final HDBProcedureSynchronizationArtefactType procedureSynchronizationArtefactType;
+    private final XSKHDBProcedureLogger procedureLogger;
+
+
+    public XSKHDBProcedureParser(XSKDataStructuresSynchronizer dataStructuresSynchronizer,
+                                 HDBProcedureSynchronizationArtefactType procedureSynchronizationArtefactType,
+                                 XSKHDBProcedureLogger procedureLogger) {
+        this.dataStructuresSynchronizer = dataStructuresSynchronizer;
+        this.procedureSynchronizationArtefactType = procedureSynchronizationArtefactType;
+        this.procedureLogger = procedureLogger;
+    }
+
+    @Override
+    public XSKDataStructureHDBProcedureModel parse(XSKDataStructureParametersModel parametersModel)
+            throws IOException, XSKDataStructuresException, XSKArtifactParserException {
```

Comment:
Why is this throwing an `IOException`? Shouldn't we wrap it in `XSKDataStructuresException` or `XSKArtifactParserException`?